### PR TITLE
Feature/allow inheritance of object subclass

### PR DESCRIPTION
**Pull request by shukob**
Allow subclassing `Object` subclass.

### CODE REVIEW

File: Pring/Document.swift [hunk 1]
```diff
@@ -231,6 +230,24 @@ public extension Document where Self: Object {
             deleteContainer.wait(block)
         }
     }
+    
+    var allChildrenUpToRootObject: [Mirror.Child]{
```
Comment (1amageek) [inline Pring/Document.swift]: {

File: Pring/Document.swift [hunk 2]
```diff
@@ -231,6 +230,24 @@ public extension Document where Self: Object {
...
+        get{
+            var mirror = Mirror(reflecting: self)
+            var children = Array(mirror.children)
+            while true{
```
Comment (1amageek) [inline Pring/Document.swift]: {

File: Pring/Object.swift [hunk 3]
```diff
@@ -152,9 +161,14 @@ open class Object: NSObject, Document {
 
         self.id = id
         self.reference = type(of: self).reference.document(id)
+        initializeWithValues(value)
+        self.isSaved = true
+    }
+    
+    
+    private func initializeWithValues(_ value: [String: Any]){
```
Comment (1amageek) [inline Pring/Object.swift]: {

File: PringTests/NestedItem.swift [hunk 4]
```diff
@@ -11,6 +11,11 @@ import Pring
 import Firebase
 import FirebaseFirestore
 
+@objcMembers
+class ChildItem: Object{
```
Comment (1amageek) [inline PringTests/NestedItem.swift]: {

**Comment by 1amageek** [comment]
Great job👏🏻

**Comment by shukob** [comment]
Hi, thanks for the review. Fixed in the latest commit.

**Comment by 1amageek** [comment]
@shukob Thank you for your PR.
Please PR to the develop branch.

I am sorry I have not prepared it.

**Comment by shukob** [comment]
Hi, PR destination had been changed. Thank you.

**Pull request merged**